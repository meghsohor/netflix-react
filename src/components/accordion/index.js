import React, { createContext, useContext, useState } from 'react';
import { Container, Inner, Title, Item, Header, Body } from './styles/accordion';

const toggleContext = createContext();

export default function Accordion({ children, ...restProps }) {
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  );
}

Accordion.Title = function AccordionTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};


Accordion.Item = function AccordionItem({ children, ...restProps }) {
  const [toggleShow, setToggleShow] = useState(false);
  return (
    <toggleContext.Provider value={{ toggleShow, setToggleShow }}>
      <Item {...restProps}>{children}</Item>
    </toggleContext.Provider>
  );
};

Accordion.Header = function AccordionHeader({ children, ...restProps }) {
  const { toggleShow, setToggleShow } = useContext(toggleContext);

  return (
    <Header onClick={() => setToggleShow((toggleShow) => !toggleShow)} {...restProps}>
      {children}
      {/* We can check a state in following way */}
      {/* <pre>{JSON.stringify(toggleShow)}</pre> */}
      {toggleShow ? (
        <img src="/images/icons/close-slim.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  );
};

Accordion.Body = function AccordionBody({ children, ...restProps }) {
  const { toggleShow } = useContext(toggleContext);

  /* return toggleShow ? <Body {...restProps}>{children}</Body> : null; */
  return (
    <Body className={toggleShow ? 'open' : 'closed'} {...restProps}>
      <span>{children}</span>
    </Body>
  );
};
