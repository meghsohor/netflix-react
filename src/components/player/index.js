import React, { useState, useContext, createContext } from 'react';
import ReactDOM from 'react-dom';
import { Container, Button, Outer, Overlay, Inner, Close } from './styles/player';

export const PlayerContext = createContext();

export default function Player({ children, ...restProps }) {
  const [showPlayer, setShowPlayer] = useState(false);

  return (
    <PlayerContext.Provider value={{ showPlayer, setShowPlayer }}>
      <Container {...restProps}>{children}</Container>
    </PlayerContext.Provider>
  );
}

Player.Video = function PlayerVideo({ src, ...restProps }) {
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);

  return showPlayer
    ? ReactDOM.createPortal(
        <Outer data-testid="player" {...restProps}>
          <Overlay onClick={() => setShowPlayer(false)} data-testid="player-overlay" />
          <Inner>
            <video id="netflix-player" controls>
              <source src={src} type="video/mp4" />
            </video>
            <Close onClick={() => setShowPlayer(false)} data-testid="player-close" />
          </Inner>
        </Outer>,
        document.body
      )
    : null;
};

Player.Button = function PlayerButton({ ...restProps }) {
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);

  return (
    <Button onClick={() => setShowPlayer((showPlayer) => !showPlayer)} {...restProps}>
      Play
    </Button>
  );
};
