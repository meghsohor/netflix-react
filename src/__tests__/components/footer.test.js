import React from 'react';
import { render } from '@testing-library/react';
import { Footer } from '../../components';

describe('<Footer />', () => {
  it('renders the <Footer /> with populated data', () => {
    const { container, getByText } = render(
      <Footer>
        <Footer.Title>Questions? Contact us.</Footer.Title>
        <Footer.Break />
        <Footer.Row>
          <Footer.Column>
            <Footer.Link href="#" onClick={(e) => e.preventDefault()}>
              FAQ
            </Footer.Link>
            <Footer.Link href="#" onClick={(e) => e.preventDefault()}>
              Investor Relations
            </Footer.Link>
            <Footer.Link href="#" onClick={(e) => e.preventDefault()}>
              Ways to Watch
            </Footer.Link>
            <Footer.Link href="#" onClick={(e) => e.preventDefault()}>
              Corporate Information
            </Footer.Link>
            <Footer.Link href="#" onClick={(e) => e.preventDefault()}>
              Netflix Originals
            </Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#" onClick={(e) => e.preventDefault()}>
              Help Centre
            </Footer.Link>
            <Footer.Link href="#" onClick={(e) => e.preventDefault()}>
              Jobs
            </Footer.Link>
            <Footer.Link href="#" onClick={(e) => e.preventDefault()}>
              Terms of Use
            </Footer.Link>
            <Footer.Link href="#" onClick={(e) => e.preventDefault()}>
              Contact Us
            </Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#" onClick={(e) => e.preventDefault()}>
              Account
            </Footer.Link>
            <Footer.Link href="#" onClick={(e) => e.preventDefault()}>
              Redeem gift cards
            </Footer.Link>
            <Footer.Link href="#" onClick={(e) => e.preventDefault()}>
              Privacy
            </Footer.Link>
            <Footer.Link href="#" onClick={(e) => e.preventDefault()}>
              Speed Test
            </Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#" onClick={(e) => e.preventDefault()}>
              Media Centre
            </Footer.Link>
            <Footer.Link href="#" onClick={(e) => e.preventDefault()}>
              Buy gift cards
            </Footer.Link>
            <Footer.Link href="#" onClick={(e) => e.preventDefault()}>
              Cookie Preferences
            </Footer.Link>
            <Footer.Link href="#" onClick={(e) => e.preventDefault()}>
              Legal Notices
            </Footer.Link>
          </Footer.Column>
        </Footer.Row>

        <Footer.Break />
        <Footer.Text>Netflix United Kingdom</Footer.Text>
      </Footer>
    );

    expect(getByText('Questions? Contact us.')).toBeTruthy();
    expect(getByText('FAQ')).toBeTruthy();
    expect(getByText('Investor Relations')).toBeTruthy();
    expect(getByText('Ways to Watch')).toBeTruthy();
    expect(getByText('Corporate Information')).toBeTruthy();
    expect(getByText('Netflix Originals')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
