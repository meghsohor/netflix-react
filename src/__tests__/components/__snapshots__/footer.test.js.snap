// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders the <Footer /> with populated data 1`] = `
<div
  class="footer__Container-qu6i6-0 kpDYGQ"
>
  <p
    class="footer__Title-qu6i6-4 jqyOrZ"
  >
    Questions? Contact us.
  </p>
  <div
    class="footer__Break-qu6i6-6 bMqiUq"
  />
  <div
    class="footer__Row-qu6i6-2 bsOPmZ"
  >
    <div
      class="footer__Column-qu6i6-1 eTQzvA"
    >
      <a
        class="footer__Link-qu6i6-3 eLZSxX"
        href="#"
      >
        FAQ
      </a>
      <a
        class="footer__Link-qu6i6-3 eLZSxX"
        href="#"
      >
        Investor Relations
      </a>
      <a
        class="footer__Link-qu6i6-3 eLZSxX"
        href="#"
      >
        Ways to Watch
      </a>
      <a
        class="footer__Link-qu6i6-3 eLZSxX"
        href="#"
      >
        Corporate Information
      </a>
      <a
        class="footer__Link-qu6i6-3 eLZSxX"
        href="#"
      >
        Netflix Originals
      </a>
    </div>
    <div
      class="footer__Column-qu6i6-1 eTQzvA"
    >
      <a
        class="footer__Link-qu6i6-3 eLZSxX"
        href="#"
      >
        Help Centre
      </a>
      <a
        class="footer__Link-qu6i6-3 eLZSxX"
        href="#"
      >
        Jobs
      </a>
      <a
        class="footer__Link-qu6i6-3 eLZSxX"
        href="#"
      >
        Terms of Use
      </a>
      <a
        class="footer__Link-qu6i6-3 eLZSxX"
        href="#"
      >
        Contact Us
      </a>
    </div>
    <div
      class="footer__Column-qu6i6-1 eTQzvA"
    >
      <a
        class="footer__Link-qu6i6-3 eLZSxX"
        href="#"
      >
        Account
      </a>
      <a
        class="footer__Link-qu6i6-3 eLZSxX"
        href="#"
      >
        Redeem gift cards
      </a>
      <a
        class="footer__Link-qu6i6-3 eLZSxX"
        href="#"
      >
        Privacy
      </a>
      <a
        class="footer__Link-qu6i6-3 eLZSxX"
        href="#"
      >
        Speed Test
      </a>
    </div>
    <div
      class="footer__Column-qu6i6-1 eTQzvA"
    >
      <a
        class="footer__Link-qu6i6-3 eLZSxX"
        href="#"
      >
        Media Centre
      </a>
      <a
        class="footer__Link-qu6i6-3 eLZSxX"
        href="#"
      >
        Buy gift cards
      </a>
      <a
        class="footer__Link-qu6i6-3 eLZSxX"
        href="#"
      >
        Cookie Preferences
      </a>
      <a
        class="footer__Link-qu6i6-3 eLZSxX"
        href="#"
      >
        Legal Notices
      </a>
    </div>
  </div>
  <div
    class="footer__Break-qu6i6-6 bMqiUq"
  />
  <p
    class="footer__Text-qu6i6-5 gurtgw"
  >
    Netflix United Kingdom
  </p>
</div>
`;
